src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -45,4 +48,27 @@ class PetService @Autowired constructor(private val petDao: PetDao) {
         }
         return "Pet owner ID updated successfully"
     }
+
+    fun  getPetsByOwnerId(ownerId: Long, companyId: Long): List<PetData> {
+        if (ownerId < 0) {
+            throw IllegalArgumentException("Owner does not exist invalid owner id!")
+        }
+        if (ownerDao.getAllOwners(companyId).none { it.ownerId == ownerId }) {
+            throw NoSuchElementException("Owner does not exist!")
+        }
+        val pets = petDao.getPetsByOwnerId(ownerId, companyId)
+        if (pets.isEmpty()) {
+            throw NoSuchElementException("This owner does not has any pets.")
+        }
```

Comment:
why are you throwing exception in this scenario? why isn't it a valid scenario that owner won't have pets?